MPI/isx.c

```diff
@@ -728,67 +740,55 @@ static void print_run_info(FILE * fp)
 
 /*
  * Prints all of the timining information for an individual PE as a row
- * to the file specificed by 'fp'. 
+ * to the file specificed by 'fp'.
  */
 static void print_timer_values(FILE * fp)
 {
-  unsigned int num_records = NUM_PES * NUM_ITERATIONS; 
+  unsigned int num_records = NUM_PES * NUM_ITERATIONS;
 
   for(unsigned int i = 0; i < num_records; ++i) {
     for(int t = 0; t < TIMER_NTIMERS; ++t){
-      if(timers[t].all_times != NULL){
-        fprintf(fp,"%f\t", timers[t].all_times[i]);
+      if(timers[t].seconds_iter > 0){
+        fprintf(fp,"%f\t", timers[t].seconds[t]);
       }
-      if(timers[t].all_counts != NULL){
-        fprintf(fp,"%u\t", timers[t].all_counts[i]);
+      if(timers[t].count_iter > 0){
+        fprintf(fp,"%u\t", timers[t].count[t]);
       }
     }
     fprintf(fp,"\n");
   }
 }
 
-/* 
+/*
  * Aggregates the per PE timing information
- */ 
+ */
 static double * gather_rank_times(_timer_t * const timer)
 {
   if(timer->seconds_iter > 0) {
 
-    const unsigned int num_records = NUM_PES * timer->seconds_iter;
+    double my_average;
+    double * restrict all_averages = NULL;
+    if (my_rank == ROOT_PE) {
+      all_averages = malloc( NUM_PES * sizeof(double));
+    }
 
-    double * restrict const all_times = malloc( num_records * sizeof(double));
+    double temp = 0.0;
+    for(int i = 0; i < timer->seconds_iter; ++i) {
+      temp += timer->seconds[i];
+    }
+    my_average = temp/(timer->seconds_iter);
 
-    MPI_Allgather(timer->seconds, timer->seconds_iter, MPI_DOUBLE,
-                  all_times, timer->seconds_iter, MPI_DOUBLE,
-                  MPI_COMM_WORLD);
+    MPI_Gather(&my_average, 1, MPI_DOUBLE,
```

Comment:
Minor spacing suggestion (ie, lining up the args):
```suggestion
    MPI_Gather(&my_average, 1, MPI_DOUBLE,
               all_averages, 1, MPI_DOUBLE,
               ROOT_PE, MPI_COMM_WORLD);
```